test/e2e/Tests/Tests/PurgeInstancesTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Durable.Tests.DotnetIsolatedE2E;
+
+[Collection(Constants.FunctionAppCollectionName)]
+public class PurgeInstancesTests
+{
+    private readonly FunctionAppFixture _fixture;
+    private readonly ITestOutputHelper _output;
+
+    public PurgeInstancesTests(FunctionAppFixture fixture, ITestOutputHelper testOutputHelper)
+    {
+        _fixture = fixture;
+        _fixture.TestLogs.UseTestLogger(testOutputHelper);
+        _output = testOutputHelper;
+    }
+
+    // Due to some kind of asynchronous race condition in XUnit, when running these tests in pipelines,
+    // the output may be disposed before the message is written. Just ignore these types of errors for now. 
+    private void WriteOutput(string message)
+    {
+        try
+        {
+            _output.WriteLine(message);
+        }
+        catch
+        {
+            // Ignore
+        }
+    }
+
+    [Theory]
+    [InlineData("PurgeOrchestrationHistory", HttpStatusCode.OK, @"^Purged [0-9]* records$")]
+    public async Task HttpTriggerTests(string functionName, HttpStatusCode expectedStatusCode, string responseRegex)
+    {
+        using HttpResponseMessage response = await HttpHelpers.InvokeHttpTrigger(functionName, "");
+        string actualMessage = await response.Content.ReadAsStringAsync();
+        Assert.Matches(responseRegex, actualMessage);
+        Assert.Equal(expectedStatusCode, response.StatusCode);
+    }
+}
```

Comment:
Are there any other test cases that we can add? Maybe a test to cover the changes you made in ProtobufUtils.cs?

Reply:
This test actually does cover that case - starttime is null in the function app code for this test, and before my change, it threw instead of returning. 
Would love to add more tests, though. I think a good set to start with would be 
- A test with empty end date
- A test with a start and end date
- Purging from (now + 1 minute) to (now + 1 day) should delete 0 records
- Starting an orchestration, then purging from >1 day ago should not purge the new orchestration's history
- Starting an orchestration, waiting for it to complete, then purging all history should delete that orchestration's history

Does this seem reasonable? Other scenarios I should cover? @bachuv @cgillum

Reply:
That sounds good to me. For the last three bullets, I remember Chris has a sample test for reference, I also referred to this when adding purge test at DTS

https://github.com/microsoft/durabletask-mssql/blob/ee4a99c31eff04d8f9595fcfeb5b5dd0a873a99f/test/DurableTask.SqlServer.AzureFunctions.Tests/CoreScenarios.cs#L311

Reply:
This sounds reasonable to me!